lib/auth/auth.go

```diff
@@ -1462,6 +1461,9 @@ func (req *GenerateServerKeysRequest) CheckAndSetDefaults() error {
 	if len(req.Roles) != 1 {
 		return trace.BadParameter("expected only one system role, got %v", len(req.Roles))
 	}
+	if len(req.PublicSSHKey) == 0 || len(req.PublicTLSKey) == 0 {
```

Comment:
Note to reviewers: the code always failed if this case were true, but it failed due to trying to parse an empty PEM block. Moving this check up into `CheckAndSetDefaults` just makes the error easier to diagnose.